File: backend/src/main/java/com/revature/models/Purchase.java
package com.revature.models;

import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "purchases")
public class Purchase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int purchaseId;

    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "orderId", referencedColumnName = "orderId")
    private Order order;

    @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JoinColumn(name="productId", referencedColumnName = "id")
    private Product product;
    private int quantity;
    private float pricePerItem;

}

File: frontend/src/app/models/purchase.ts
import { Order } from "./order";
import { Product } from "./product";

export class Purchase {
    purchaseId: number;
    order: Order;
    product: Product;
    quantity: number;
    pricePerItem: number;

    constructor (purchaseId: number, order: Order, product: Product, quantity: number, pricePerItem: number) {
        this.purchaseId = purchaseId;
        this.order = order;
        this.product = product;
        this.quantity = quantity;
        this.pricePerItem = pricePerItem;
    }
}